Add due diligence guide (#98)
wealthkernel/api-documentation-guides: docs/due-diligence/Basics.md

---
tags: [Due Diligence]
---

# Due Diligence Outsourcing

We require KYC checks to be performed on parties as part of the onboarding process. Regular checks must also be in place as long as the party is active in the system. Our Due Diligence Outsourcing model is evidence-based, meaning that you can provide evidence for a KYC check along with the outcome from an external KYC provider or your own KYC integration.

Please note that Due Diligence Outsourcing is only available for persons. Organizations require a different set of checks.

> ### Expectations for tenants using due diligence outsourcing

Author: I changed it to a h2

Check outcomes and monitoring information can be added to a person's due diligence summary through a due diligence update. Any of the following due diligence fields can be updated:

- Due diligence level
- Identity
- Politically exposed persons (PEPs)
- Sanctions
- Material information

The due diligence update doesn't need all of this information at once. Partial updates are also supported. In the due diligence update request, any undefined field within the five fields mentioned above will be ignored. For example, the following update request is valid:

```json
{
  "partyId": "pty-123",
  "updatedBy": "employee@company.com",
  "politicalExposure": {
    "continuousCheckStartedOn": "2025-04-08",
    "continuousCheckStatus": "Active"
  }
}
```

This request will only update the PEP's continuous monitoring information, without altering the PEP status, or any other check information.

A `null` value, in a nullable field, will be honored and recorded in the due diligence summary, for example:

```json
{
  "partyId": "pty-123",
  "updatedBy": "employee@company.com",
  "politicalExposure": {
    "continuousCheckStartedOn": null,
    "continuousCheckStatus": "Inactive"
  }
}

> ```

Author: The suggestion looked odd due to the double end of snippet "```". I added the last paragraph you suggested

Once the evidence definition is in place, you must upload the corresponding content; otherwise, the due diligence update cannot be submitted.

### Lifecycle

The due diligence update has the following lifecycle:

```mermaid
stateDiagram-v2
    direction LR
    [*] --> Pending
    Pending --> Submitted : Submit
    Submitted --> Completed
    Submitted --> Rejected
```

When a due diligence update is created, it stays as `Pending`. Note that it is only possible to add evidence to a due diligence update in this state. After submitting the update, it will be automatically completed. However, the update could be rejected under certain circumstances, for example, when the provided `PartyId` does not exist.

The diagram below shows a typical workflow of a due diligence update.

```mermaid
sequenceDiagram
    autonumber
    participant Tenant
    participant WealthKernel

    Tenant->>WealthKernel: POST /parties (Person type)
    WealthKernel->>WealthKernel: Initialize the Person due diligence summary
    
    Tenant->>WealthKernel: POST /due-diligence-update/person

>     activate WealthKernel

Author: It indicates the start and end of the update process, it shows it as a block in WealthKernel's lifeline

    loop
    Tenant->>WealthKernel: POST /due-diligence-update/person/perddu-123/evidence
    Tenant->>WealthKernel: POST /due-diligence-update/person/evidence/perdduevi-111/contents
    end
    Tenant->>WealthKernel: POST /due-diligence-update/person/perddu-123/actions/submit
    WealthKernel->>WealthKernel: Update the Person due diligence summary
    deactivate WealthKernel
```

Steps 3 to 6 show the requests needed to add a due diligence update, from initializing the due diligence update to submitting it.

> Steps 4 and 5 can be repeated as many times as needed, as it is allowed to include multiple evidence items in an update, including none. Note the different IDs used in these requests. Step 4 uses the due diligence update ID, whereas Step 5 uses the evidence ID.

Author: I see what you mean, I removed that bit

---
tags: [Due Diligence]
---

# Due Diligence Outsourcing

> We require KYC checks to be performed for each party as part of the onboarding process. Regular checks must also be in place as long as the party has active accounts in the system. Our due diligence outsourcing model is evidence-based, meaning that you can provide evidence for a KYC check along with the outcome from an external KYC provider or your own KYC integration.

Reviewer: ```suggestion
We require KYC checks to be performed for each party as part of the onboarding process. Regular checks must also be in place as long as the party has active accounts in the system. Our due diligence outsourcing model is evidence-based, meaning that you should provide evidence for a KYC check along with the outcome from an external KYC provider or your own KYC integration.
```

The diagram below shows a typical workflow of a due diligence update.

```mermaid
sequenceDiagram
    autonumber
    participant Tenant
    participant WealthKernel

    Tenant->>WealthKernel: POST /parties (Person type)
    WealthKernel->>WealthKernel: Initialize the Person due diligence summary
    
    Tenant->>WealthKernel: POST /due-diligence-update/person
    activate WealthKernel
    loop
    Tenant->>WealthKernel: POST /due-diligence-update/person/perddu-123/evidence
    Tenant->>WealthKernel: POST /due-diligence-update/person/evidence/perdduevi-111/contents
    end
    Tenant->>WealthKernel: POST /due-diligence-update/person/perddu-123/actions/submit
    WealthKernel->>WealthKernel: Update the Person due diligence summary
    deactivate WealthKernel
```

Steps 3 to 6 show the requests needed to add a due diligence update, from initializing the due diligence update to submitting it.

> Steps 4 and 5 can be repeated as many times as needed, as it is allowed to include multiple evidence items in an update, including none.

Reviewer: ```suggestion
Steps 4 and 5 can be repeated as many times as needed if there is evidence to upload, as it is allowed to include multiple evidence items in an update.
```

> Finally, when the due diligence update completes in Step 6, the due diligence summary will apply the update.

Reviewer: ```suggestion
Finally, when the due diligence update completes in Step 6, the update will be applied to the due diligence summary.
```

> ## Due diligence and Accounts

Reviewer: ```suggestion
## Due diligence and accounts
```

Reviewer: Just for consistency with the others
